### DISCUSSION TITLE
How do I permanently hide the Dock and statusbar?

### DISCUSSION THREAD

**Discussion by hnrq** [asmvik/yabai#854] [General]
Hi all! I've been using BSPWM for a couple of years since I started working. However, by company policies, I have to use a Mac now :(. Yabai makes me feel at home, but the statusbar and the Dock annoys me (Cuz they're really ugly IMO lol). Anyone knows how do I permanently hide them?

**Comment by dominiklohmann** [comment]
<img width="780" alt="image" src="https://user-images.githubusercontent.com/4488655/110172475-7ea6f180-7dfd-11eb-88a8-43acade499ad.png">

Check the boxes `Automatically hide and show the Dock` and `Automatically hide and show the menu bar`. That's the best you can do.

**Reply by hnrq** [reply]
Really? Oh, well, that's bad :/ 
Thanks anyway!

**Reply by marconetto** [reply]
Hi @hnrq 

I've faced the same problem, because even if I hide it, sometimes I would pass the mouse closed to the region of the menubar and it would show it up... really annoying.

I would suggest you explore hammerspoon.  Yabai+hammerspoon+karabiner+Übersicht working together can help you do interesting stuff with your mac. Take a look at this thread in hammerspoon on how to achieve what you want:
https://groups.google.com/g/hammerspoon/c/xQ7AjexL31o

This is my current menubar (it looks ugly, but it has everything I need) :)
![image](https://user-images.githubusercontent.com/7735345/119001905-c8b26480-b962-11eb-9a02-64feb1b0de5b.png)

**Comment by anuragrao04** [comment]
you can set the dock show delay to something arbitrarily high:

```zsh
defaults write com.apple.dock autohide -bool true
defaults write com.apple.dock autohide-delay -float 1000
defaults write com.apple.dock no-bouncing -bool TRUE
killall Dock
```

The `no-bouncing` property is used to avoid the icons bouncing and showing on the screen when an app is launching or there it has a new pop up. 

Unfortunately, this works when yabai is off but doesn't when yabai is on. Yabai does something with the Dock to make it appear faster on screen when hovered over. This overrides my above modifications. I have opened an issue about the same: #1951

**Reply by hnrq** [reply]
Fortunately, the destiny freed me from the OSX curse, so I can now use Linux again

**Comment by anuragrao04** [comment]
Did you get a new computer or are you talking about Asahi Linux?

**Reply by hnrq** [reply]
Nah, I sold the Macbook because I got a new job that didn't need it

**Comment by asmvik** [comment]
In latest versions of yabai you can hide the menubar completely: `yabai -m config menubar_opacity 0.0`

**Reply by shigeosapsycho** [reply]
can you hide the dock instead of just the menubar?

**Reply by matthewshammond** [reply]
Is there a way to query the menubar_opacity?

Trying to bind a skhd hotkey to set menubar_opacity to 1.0 if it is 0.0 and set to 0.0 if it is 1.0.
